### PR TITLE
[Bug fix] Add conditional bosses filter if gamemode is seasonal

### PR DESCRIPTION
**Pull request by Not-Jayden**
Fix for boss results being returned incorrectly as mentioned in issue #34

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -4,7 +4,12 @@
     "module": "commonjs",
     "declaration": true,
     "outDir": "./lib",
-    "strict": true
+    "strict": true,
+    "lib": ["ES2015", "DOM", "DOM.Iterable"],
+    "typeRoots": [
+      "./node_modules/@types",
+      "./src/@types"
+    ]
```
Comment (Not-Jayden) [inline tsconfig.json:12]: VSCode was giving me a bunch of type errors without these inclusions.

File: src/hiscores.ts [hunk 2]
```diff
@@ -97,11 +97,16 @@ export function parseStats(csv: string): Stats {
       return activity;
     });
 
+  const filteredBosses =
+    mode === 'seasonal'
+      ? BOSSES.filter((boss) => boss !== 'tempoross')
+      : BOSSES;
```
Comment (Not-Jayden) [inline src/hiscores.ts:103]: The main fix. Kind of annoying it has to handle the one edge case here but it is what it is :/

**Comment by maxswa** [comment]
First of all, thank you for finding and fixing this edge case! Would you be comfortable adding a new unit test to account for this behavior?

File: src/hiscores.ts (hunk 2 above)
Comment (maxswa) [reply src/hiscores.ts:103]: Yeah that is a strange edge case. Can you add a comment above `filteredBosses` in order to explain the behavior? Plus it will flag it for future removal once they fix this on their side.

Comment (maxswa) [reply src/hiscores.ts:103]: After verifying this edge case myself, I believe the missing boss is actually the newly added `theatreOfBloodHardMode`. If you compare the stats of `Fysad` in seasonal (on the left) to the normal game:

image

Line 70 on both sides is The Gauntlet, while Zulrah is the last line for both sides. Since The Gauntlet is below Tempoross on the list, it has to be another boss which is omitted.
Then I visited the hiscores page, and clicked on "Leagues" and sure enough TOB: Hard Mode was missing from the side bar:
https://secure.runescape.com/m=hiscore_oldschool_seasonal/overall

image

Comment (Not-Jayden) [reply src/hiscores.ts:103]: Oh shoot you're right 🤦‍♂️ I'll update this after work today.
